content/articles/assertion-and-design-by-contract-in-java/index.md

```diff
@@ -0,0 +1,153 @@
+
+An `AssertionError` exception is thrown once the expression is false and contains details of the bug. That is, the line number and file in which the errors are in the program. To implement assertions in Java we use the assert statement. The assert statement can be implemented in two ways.
+
+```Java
+assert Expression 1;
+```
+The expression is a boolean, if the boolean report is false The assertion error is thrown without information about bugs that happened in the program.
+
+```Java
+assert Expression 1; Expression 2;
+```
+
+The first expression is boolean and the second is the value passed to the Exception of the `AssertError`. We can use the assert statement as:
+
+```Java
+public void division()
+{
+    double a=b/c;  ///c should not be equal to zero
+}
+``` 
+
+The above comment can be replaced using the assert statement:
+```Java
+public void division()
+{
+    assert c!=0  /// this assert statement states that c should not be equal to zero
+    double a=b/c
+}
+```
+
+If the expression of the assert statement is false, an error is thrown specifying that the program has errors. This is referred to as `assertion failure`.
+
+### Enabling and disabling assertions
+Java provides command line parameters to allow enabling and disabling assertions. Assertions enabling is done using the following parameters:
+
+```Java
+-ea
+or 
+-enableassertions
+``` 
+
+Enabling assertions from a file Myfile.java we use the command:
+
+```Java
+java -enableassertions Myfile.java
+```
+or
+
+```Java
+java -ea Myfile.java
+```
+Disabling assertions  we use the command-line parameter:
+
+```Java
+-da
+or
+disableassertions
+```
+Disabling assertions from a file Myfile.java we use the command:
+
+```Java
+java -disableassertions Myfile.java
+```
+
+or
+
+```Java
+java -da Myfile.java
+```
+
+### Command lines to enable assertions
+1. `Java -ea`- This command-line is enabled in all classes except the system classes.
+2. `Java -ea Main`- This command-line enables assertions for all the main classes.
+3. `Java -eaTestClass Main`- This command-line enables the only assertion in one class -`TestClass` in the main program.
+4. `Java -ea...Main`- This command is used to enable assertions for the working class in the main directory
+
+### Assertion Rules
+Since assertion is used to check on the validation of assumptions made by the programmer at the time of execution. Then, some rules govern the use of assertions:
+
+1. Checking the method arguments passed.
+2. Using the assertion in the default case of the switch statement.
+3. Making use of an assertion descriptive.
+4. Avoid processing in an assertion condition.
+5. Avoid catching an assertion-related exception.
+6. Avoid the use of more than one condition in an assertion
+
+### Creating a Java program
+This program helps the learner to implement assertion by dividing two numbers.
```

Comment:
Here, we need to add some comments in the program and an explanation of what is happening after the program